src/hotspot/share/runtime/stubDeclarations.hpp

```diff
@@ -165,17 +171,1121 @@
+  do_stub(compiler, sha256_implCompress)                                \
+  do_entry(compiler, sha256_implCompress, sha256_implCompress,          \
+           sha256_implCompress)                                         \
+  do_stub(compiler, sha256_implCompressMB)                              \
+  do_entry(compiler, sha256_implCompressMB, sha256_implCompressMB,      \
+           sha256_implCompressMB)                                       \
+  do_stub(compiler, sha512_implCompress)                                \
+  do_entry(compiler, sha512_implCompress, sha512_implCompress,          \
+           sha512_implCompress)                                         \
+  do_stub(compiler, sha512_implCompressMB)                              \
+  do_entry(compiler, sha512_implCompressMB, sha512_implCompressMB,      \
+           sha512_implCompressMB)                                       \
+  do_stub(compiler, sha3_implCompress)                                  \
+  do_entry(compiler, sha3_implCompress, sha3_implCompress,              \
+           sha3_implCompress)                                           \
+  do_stub(compiler, sha3_implCompressMB)                                \
+  do_entry(compiler, sha3_implCompressMB, sha3_implCompressMB,          \
+           sha3_implCompressMB)                                         \
+  do_stub(compiler, updateBytesAdler32)                                 \
+  do_entry(compiler, updateBytesAdler32, updateBytesAdler32,            \
+           updateBytesAdler32)                                          \
+  do_stub(compiler, multiplyToLen)                                      \
+  do_entry(compiler, multiplyToLen, multiplyToLen, multiplyToLen)       \
+  do_stub(compiler, squareToLen)                                        \
+  do_entry(compiler, squareToLen, squareToLen, squareToLen)             \
+  do_stub(compiler, mulAdd)                                             \
+  do_entry(compiler, mulAdd, mulAdd, mulAdd)                            \
+  do_stub(compiler, montgomeryMultiply)                                 \
+  do_entry(compiler, montgomeryMultiply, montgomeryMultiply,            \
+           montgomeryMultiply)                                          \
+  do_stub(compiler, montgomerySquare)                                   \
+  do_entry(compiler, montgomerySquare, montgomerySquare,                \
+           montgomerySquare)                                            \
+  do_stub(compiler, bigIntegerRightShiftWorker)                         \
+  do_entry(compiler, bigIntegerRightShiftWorker,                        \
+           bigIntegerRightShiftWorker, bigIntegerRightShift)            \
+  do_stub(compiler, bigIntegerLeftShiftWorker)                          \
+  do_entry(compiler, bigIntegerLeftShiftWorker,                         \
+           bigIntegerLeftShiftWorker, bigIntegerLeftShift)              \
+  AARCH64_ONLY(                                                         \
```

Comment:
It's unfortunate to see platform-specific declarations in `stubDeclarations.hpp`. Any chances to keep them in CPU-specific files (as it is shaped now)?

Reply:
Latest version has arch declarations in arch-specific files